src/project-converter.ts

```diff
@@ -43,41 +35,74 @@ const excludeFromResults = new Set([
  * constructor.
  */
 export class ProjectConverter {
+  readonly analysis: Analysis;
   readonly urlHandler: UrlHandler;
   readonly conversionSettings: ConversionSettings;
 
+  private readonly convertedPackages = new Set<string>();
+
   /**
    * A cache of all JS Exports by namespace, to map implicit HTML imports to
    * explicit named JS imports.
    */
   readonly namespacedExports = new Map<string, JsExport>();
+  readonly results = new Map<
+      OriginalDocumentUrl,
+      JsModuleConversion|DeleteFileConversion|HtmlDocumentConversion>();
+
+  constructor(
+      analysis: Analysis, urlHandler: UrlHandler,
+      conversionSettings: ConversionSettings) {
+    this.analysis = analysis;
+    this.urlHandler = urlHandler;
+    this.conversionSettings = conversionSettings;
+  }
 
   /**
-   * A cache of all converted documents. Document conversions should be
-   * idempotent, so conversion results can be safely cached.
+   * Convert a document and any of its dependencies. The output format (JS
+   * Module or HTML Document) is determined by whether the file is included in
+   * conversionSettings.includes.
    */
-  readonly conversionResults = new Map<OriginalDocumentUrl, ConversionResult>();
+  convertPackage(matchPackageName: string) {
+    this._convertPackage(matchPackageName, new Set());
+  }
 
-  constructor(urlHandler: UrlHandler, conversionSettings: ConversionSettings) {
-    this.urlHandler = urlHandler;
-    this.conversionSettings = conversionSettings;
+  /**
+   * Convert a document and any of its dependencies. The output format (JS
+   * Module or HTML Document) is determined by whether the file is included in
+   * conversionSettings.includes.
+   */
+  private _convertPackage(
+      matchPackageName: string, visitedPackages: Set<string>) {
+    const htmlDocuments =
+        [...this.analysis.getFeatures({kind: 'html-document'})].filter((d) => {
+          const documentUrl = this.urlHandler.getDocumentUrl(d);
+          const packageName =
+              this.urlHandler.getOriginalPackageNameForUrl(documentUrl);
+          return packageName === matchPackageName;
+        });
```

Comment:
I would sort the files first for deterministic ordering: first by directory depth, then alpha.

Reply:
Actually, I would add a document-order traversal from that list to get the final ordering...